packages/core/lib/__tests__/use-metadata.spec.tsx

```diff
@@ -0,0 +1,172 @@
+import { act, cleanup, renderHook, waitFor } from "@testing-library/react";
+import { AppState, Config, Data, MetaData } from "../../types";
+import { useResolvedData } from "../use-resolved-data";
+import { SetAction, SetDataAction } from "../../reducer";
+import { cache } from "../resolve-component-data";
+import { defaultAppState } from "../../components/Puck/context";
+import { RefreshPermissions } from "../use-resolved-permissions";
+
+const item1 = { type: "MyComponent", props: { id: "MyComponent-1" } };
+const item2 = { type: "MyComponent", props: { id: "MyComponent-2" } };
+const item3 = { type: "MyComponent", props: { id: "MyComponent-3" } };
+
+const data: Data = {
+  root: { props: { title: "" } },
+  content: [item1],
+  zones: {
+    "MyComponent-1:zone": [item2],
+    "MyComponent-2:zone": [item3],
+  },
+};
+
+const metadata: MetaData = {
+  "title": "Hello, Metadata!"
+}
+
+const state: AppState = {
+  data,
+  ui: defaultAppState.ui,
+};
+
+const config: Config = {
+  root: {
+    resolveData: (data) => {
+      return {
+        ...data,
+        props: { title: "Resolved title" },
+        readOnly: { title: true },
+      };
+    },
+  },
+  components: {
+    MyComponent: {
+      defaultProps: { prop: "example" },
+      resolveData: ({ props }, { metadata }) => {
+        return {
+          props: {
+            ...props,
+            prop: metadata.title || "Hello, world",
+          },
+          readOnly: {
+            prop: true,
+          },
+        };
+      },
+      render: () => <div />,
+    },
+  },
+};
+
+describe("use-metadata", () => {
```

Comment:
There is no `use-metadata` hook implemented, so I suspect these would be better added to the `useResolveData` tests

Reply:
I've updated the tests but an additional review would be best.